File: src/main/kotlin/com/riskrieg/mapeditor/model/map/graph/Territory.java
package com.riskrieg.mapeditor.model.map.graph;

import java.awt.Point;
import java.util.HashSet;
import java.util.Objects;
import java.util.Set;

public class Territory implements Comparable<Territory> { // TODO: Convert to Record when Java 16 comes out

  private final String name;
  private final Set<Point> seedPoints;

  public Territory(String name, Set<Point> seedPoints) {
    Objects.requireNonNull(name);
    Objects.requireNonNull(seedPoints);
    if (seedPoints.isEmpty()) {
      throw new IllegalStateException("seedPoints must not be empty");
    }
    this.name = name;
    this.seedPoints = seedPoints;
  }

  public Territory(String name, Point seedPoint) {
    Objects.requireNonNull(name);
    Objects.requireNonNull(seedPoint);
    this.name = name;
    this.seedPoints = new HashSet<>();
    this.seedPoints.add(seedPoint);
  }

  public String name() {
    return name;
  }

  public Set<Point> seedPoints() {
    return seedPoints;
  }

  @Override
  public int compareTo(Territory o) {
    return this.name().compareTo(o.name());
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Territory territory = (Territory) o;
    return Objects.equals(name, territory.name) &&
        Objects.equals(seedPoints, territory.seedPoints);
  }

  @Override
  public int hashCode() {
    return Objects.hash(name, seedPoints);
  }

  @Override
  public String toString() {
    return name;
  }

}

File: src/main/kotlin/com/riskrieg/mapeditor/model/Territory.kt
package com.riskrieg.mapeditor.model

import java.awt.Point
import java.util.*

class Territory(val name: String, val seedPoints: Set<Point> = HashSet()) {

    constructor(name: String, seedPoint: Point) : this(name, setOf<Point>(seedPoint))

    override fun equals(o: Any?): Boolean {
        if (this === o) {
            return true
        }
        if (o == null || javaClass != o.javaClass) {
            return false
        }
        val territory = o as Territory
        return name == territory.name && seedPoints == territory.seedPoints
    }

    override fun hashCode(): Int {
        return Objects.hash(name, seedPoints)
    }

    init {
        require(name.isNotBlank()) { "String 'name' cannot be blank" }
        check(seedPoints.isNotEmpty()) { "Set<Point> 'seedPoints' must not be empty" }
    }
}